File: VB/CustomDescendants/MyBandedGridView.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid.Customization
Imports System.ComponentModel
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.Utils

Namespace TreeListCustomization
	Friend Class MyBandedGridView
		Inherits BandedGridView
		' constructors
		Public Sub New()
			Me.New(Nothing)
		End Sub
		Public Sub New(ByVal grid As GridControl)
			MyBase.New(grid)
		End Sub

		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyBandedGridView"
			End Get
		End Property

		Protected Overrides Function CreateCustomizationForm() As CustomizationForm
			Return New TreeListBandedCustomizationForm(Me)
		End Function

		Friend Shadows ReadOnly Property Handler() As BaseViewHandler
			Get
				Return MyBase.Handler
			End Get
		End Property
	End Class
End Namespace

File: CS/CustomDescendants/MyBandedGridView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid.Customization;
using System.ComponentModel;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.Utils;

namespace TreeListCustomization
{
    class MyBandedGridView : BandedGridView
    {
        // constructors
        public MyBandedGridView() : this(null) { }
        public MyBandedGridView(GridControl grid) : base(grid) { }

        protected override string ViewName { get { return "MyBandedGridView"; } }

        protected override CustomizationForm CreateCustomizationForm()
        {
            return new TreeListBandedCustomizationForm(this);
        }

        internal new BaseViewHandler Handler { get { return base.Handler; } }
    }
}
